compiler/rustc_target/src/spec/json.rs

```diff
@@ -116,6 +116,18 @@ impl Target {
                     Some(Ok(()))
                 })).unwrap_or(Ok(()))
             } );
+            ($key_name:ident, FloatAbi) => ( {
+                let name = (stringify!($key_name)).replace("_", "-");
+                obj.remove(&name).and_then(|o| o.as_str().and_then(|s| {
+                    match s.parse::<super::FloatAbi>() {
+                        Ok(float_abi) => base.$key_name = Some(float_abi),
+                        _ => return Some(Err(format!("'{}' is not a valid value for \
```

Comment:
Might be worth leaving a note in the error messages to users of `softfp` (the clang-level flag) as to how to get a similar effect in Rust? (I think this is just to enable the FPU in CPU features but set the float ABI to soft, but I haven't checked.)

Reply:
So `-mfloat-abi=softfp` exists as a separate value from `-mfloat-abi=soft`? Someone will have to figure out what it does then, I have no clue.

> I think this is just to enable the FPU in CPU features but set the float ABI to soft

That makes little sense to me, the flag explicitly has ABI in the same so it should solely control the ABI, completely independently of whether an FPU is actually present.

Reply:
For clang, setting `-mfloat-abi=soft` implicitly disables the FPU too. We could also just dodge this entirely for now by not mentioning `-mfloat-abi`

Reply:
[`clang` set `-msoft-float` when talking to LLVM if `-mfloat-abi=soft` is set](https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/Clang.cpp#L1640)

Reply:
Okay so they added extra mess on top of the LLVM mess, amazing...

Yeah probably better not to mention the clang flag then.